File: collection/dictionary/first_recursive.py
pattern="ABABBAC"
dict={} # use hashing technique
#Find first recursive character - first adyam repeat chaiyuna character
for ch in pattern: #print each character one by one
    print(ch)
    if ch not in dict:
        dict[ch]=1
    else:
        print(ch,"is recursive character")
        break

File: javascript/oobs_javascript/firstrecursive.js
//#Find first recursive character - first adyam repeat chaiyuna character
pattern="ABBAABBA"
var dict={}
for(let ch of pattern){
    //console.log(ch);
    
    if (ch in dict){
        console.log("reversive character",ch);
        break;
           
    }
    else{
        dict[ch]=1
        }
    
    
}
